### PR TITLE
[Form] Create new `Form` component

### PR DESCRIPTION
**Pull request by atomiks**
(Branched off the `Field` components PR in my fork. The extraneous Field commits will disappear once it's merged).

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/mui-base/src/Checkbox/Root/useCheckboxRoot.ts [hunk 1]
```diff
@@ -60,13 +62,35 @@ export function useCheckboxRoot(params: UseCheckboxRootParameters): UseCheckboxR
...
+        validityData: getCombinedFieldValidityData(validityData, invalid),
+        validate() {
+          if (!inputValidationRef.current) {
+            return;
+          }
+
+          const controlValue = inputValidationRef.current.checked;
+          markedDirtyRef.current = true;
+
+          // Syncronously update the validity state so the submit event can be prevented.
```
Comment (michaldudak) [inline packages/mui-base/src/Checkbox/Root/useCheckboxRoot.ts:87]:
```suggestion
          // Synchronously update the validity state so the submit event can be prevented.
```

File: packages/mui-base/src/Form/Root/FormRoot.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  onSubmit: PropTypes.func,
+  /**
+   * A function to customize rendering of the component.
+   */
+  render: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+} as any;
+
+export { FormRoot };
+
+namespace FormRoot {
```
Comment (michaldudak) [inline packages/mui-base/src/Form/Root/FormRoot.tsx:121]: Nit: could you move it above proptypes?

File: packages/mui-base/src/Slider/Root/useSliderRoot.ts [hunk 3]
```diff
@@ -161,15 +177,40 @@ function useSliderRoot(parameters: UseSliderParameters): UseSliderReturnValue {
     };
   }, [id, setControlId]);
 
+  useEnhancedEffect(() => {
```
Comment (michaldudak) [inline packages/mui-base/src/Slider/Root/useSliderRoot.ts:180]: This seems to be pretty much copied and pasted across all form components. Have you considered creating a hook encapsulating this logic?

File: docs/data/base/components/form/UnstyledFormIntroduction/system/index.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
```
Comment (michaldudak) [inline docs/data/base/components/form/UnstyledFormIntroduction/system/index.tsx:1]:
```suggestion
'use client';

import * as React from 'react';
```

File: packages/mui-base/src/Field/Control/FieldControl.types.ts [hunk 5]
```diff
@@ -5,4 +5,9 @@ export type FieldControlElement = HTMLInputElement | HTMLTextAreaElement | HTMLS
 
 export type FieldControlOwnerState = FieldRootOwnerState;
 
-export interface FieldControlProps extends BaseUIComponentProps<'input', FieldControlOwnerState> {}
+export interface FieldControlProps extends BaseUIComponentProps<'input', FieldControlOwnerState> {
```
Comment (michaldudak) [inline packages/mui-base/src/Field/Control/FieldControl.types.ts:8]: Move types to namespaces

File: packages/mui-base/src/Form/Root/FormRoot.test.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import * as Form from '@base_ui/react/Form';
+import * as Field from '@base_ui/react/Field';
+import { expect } from 'chai';
+import { createRenderer, fireEvent, screen } from '@mui/internal-test-utils';
+import { describeConformance } from '../../../test/describeConformance';
+
```
Comment (michaldudak) [inline packages/mui-base/src/Form/Root/FormRoot.test.tsx:7]: Import both createRenderer and describeConformance from '#test-utils'

**Comment by atomiks** [comment]
Conflict errors are too complex to resolve. I'll create a new PR.

**Pull request closed without merging**